src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
```

Comment:
I'm not against caching and will do for the cause, as said in an another comment JIT probably already does it though code-motion because the code in the loop does not mutates the array. JIT does ALOT of code motions :)

More generally on micro-benchmarks, this video is really a must see: https://www.youtube.com/watch?v=65-RbBwZQdU